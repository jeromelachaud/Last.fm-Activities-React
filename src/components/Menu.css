.Menu__item a{
  cursor: pointer;
  color: rgba(255,255,255,1);
}

.Menu__item:hover a,
.Menu__item.active a{
    background-color: rgba(255,255,255,0.4);
}
