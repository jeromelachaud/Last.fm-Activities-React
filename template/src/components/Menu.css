.Menu {
  position: fixed;
  right: 20px;
  width: auto;
  text-align: right;
  background-color: grey;
}

.Menu__list {
  text-align: right;
}

.Menu__item{
    color: rgba(255,255,255,0.6);
}

.Menu__item:hover,
.Menu__item.active{
    background-color: rgba(255,255,255,0.4);
}
