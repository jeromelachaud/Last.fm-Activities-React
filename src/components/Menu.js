import React from 'react';
import {
  Link
} from 'react-router';
import './Menu.css';
import User from './User';
import language from '../language/language';

const menuLanguage = language.components.menu;

let Menu = React.createClass ({

  getInitialState() {
    return { showUser: false };
  },

  toggleUser(e) {
    e.preventDefault(e);
    if (!this.state.showUser) {
      this.setState({showUser : true});
    } else {
      this.setState({showUser : false});
    }
  },

  render() {

    const {
      user
    } = this.props;

    let userElement;
    userElement = (
      <User
        user={user} />
    );

    return (
      <div>
        <ul>
          <li
            className="Menu__item">
            <Link
              to="recent-tracks">
              {menuLanguage.recentTracks}
            </Link>
          </li>
          <li
            className="Menu__item">
            <Link
              to="top-artists">
              {menuLanguage.topArtists}
            </Link>
          </li>
          <li
            className="Menu__item">
            <Link
              to="#0"
              onClick={this.toggleUser}>
              {menuLanguage.userInfo}
            </Link>
          </li>
        </ul>
        {this.state.showUser ? userElement : null}
      </div>
    );
  }
});

export default Menu;
