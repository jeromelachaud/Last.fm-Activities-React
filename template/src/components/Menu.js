import React from 'react';
import {
  Link
} from 'react-router';
import './Menu.css';
import User from './User';


let Menu = React.createClass ({

  handleClick: function() {
    console.log('foo');
  },

  render() {

    const {
      user
    } = this.props;

    return (
      <div>
        <ul>
          <li>
            <Link to="/recent-tracks" className="Menu__item">Latest track</Link>
          </li>
          <li>
            <a href="/top-artists" className="Menu__item">Top artists</a>
          </li>
          <li>
            <a href="#0" className="Menu__item">Charts</a>
          </li>
          <li>
            <a href="/" className="Menu__item" onClick={this.handleClick}>User infos</a>
          </li>
        </ul>
        <User
          user={user} />
      </div>
    );
  }
});

export default Menu;
