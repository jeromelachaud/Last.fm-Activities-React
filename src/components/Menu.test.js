import React from 'React';
import {
  shallow
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Menu from './Menu';

describe('Menu componemts', () => {
  it('Menu component should render as expected', () => {
    const menuComponent = shallow(<Menu />);
    expect(toJson(menuComponent)).toMatchSnapshot();
  });
});
