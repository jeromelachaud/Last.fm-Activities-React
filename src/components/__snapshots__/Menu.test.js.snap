exports[`Menu componemts Menu component should render as expected 1`] = `
<div>
  <ul>
    <li>
      <Link
        className="Menu__item"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="recent-tracks">
        Recent Tracks
      </Link>
    </li>
    <li>
      <Link
        className="Menu__item"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="top-artists">
        Top Artists
      </Link>
    </li>
    <li>
      <Link
        className="Menu__item"
        onClick={[Function]}
        onlyActiveOnIndex={false}
        style={Object {}}>
        User Info
      </Link>
    </li>
  </ul>
</div>
`;
